keywords/TestServeriOS.py

```diff
@@ -0,0 +1,323 @@
+                    log_info("Waiting for the iPhone Simulator to boot up")
+                    time.sleep(1)
+                    continue
+
+        # Get the device ID
+        list_output = subprocess.Popen(["xcrun", "simctl", "list"], stdout=subprocess.PIPE)
+        output = subprocess.check_output(('grep', 'Booted'), stdin=list_output.stdout)
+
+        for line in output.splitlines():
+            if "Phone" in line:
+                self.device_id = re.sub(' +', ' ', line).strip()
+                self.device_id = self.device_id.split(" ")[4]
+                self.device_id = self.device_id.strip('(')
+                self.device_id = self.device_id.strip(')')
+
+        if not self.device_id:
+            raise LiteServError("Could not get the device ID of the running simulator")
+
+    def remove_device(self):
+        """
+        Remove the iOS app from the connected device
+        """
+        bundle_id = "com.couchbase.CBLTestServer-iOS"
+
+        output = subprocess.check_output([
+            "ios-deploy", "--uninstall_only", "--bundle_id", bundle_id
+        ])
+        log_info(output)
+
+        # Check that removal is successful
+        output = subprocess.check_output(["ios-deploy", "--list_bundle_id"])
+        log_info(output)
+
+        if bundle_id in output:
+            raise LiteServError("CBLTestServer-iOS is still present after uninstall")
+
+    def remove(self):
+        """
+        Remove the iOS app from the simulator
+        """
+        bundle_id = "com.couchbase.CBLTestServer-iOS"
+        log_info("Removing CBLTestServer")
+
+        self.stop()
+
+        # Stop the simulator
+        log_info("device_id: {}".format(self.device_id))
+        output = subprocess.check_output([
+            "killall", "Simulator"
+        ])
+
+        # Erase the simulator
+        output = subprocess.check_output([
+            "xcrun", "simctl", "erase", self.device_id
+        ])
+
+        if bundle_id in output:
+            raise LiteServError("{} is still present after uninstall".format(bundle_id))
+
+    def start(self, logfile_name):
+        """
+        1. Starts a LiteServ with logging to provided logfile file object.
+           The running LiteServ process will be stored in the self.process property.
+        2. The method will poll on the endpoint to make sure LiteServ is available.
+        3. The expected version will be compared with the version reported by http://<host>:<port>
+        4. Return the url of the running LiteServ
+        """
+
+        # self.device = "iPhone-7-Plus"
+        self.logfile_name = logfile_name
+
+        # package_name = "CBLTestServer-iOS.app"
+        # app_dir = "CBLTestServer-iOS"
+
+        self.app_path = "{}/{}/{}".format(BINARY_DIR, self.app_dir, self.app_name)
+        # self.app_path = "/Users/sridevi.saragadam/Library/Developer/Xcode/DerivedData/CBLTestServer-iOS-ayypbvnmphhpihebbhrjusdqhzmk/Build/Products/Release-EE-iphonesimulator/CBLTestServer-iOS.app"
```

Comment:
remove the commented code, if not needed.

Reply:
done